.github/workflows/pr.yaml

```diff
@@ -68,27 +80,82 @@ jobs:
           make manifests generate fmt vet install
           make run &
 
+      - name: Download SeaLights Go agent and CLI tool
+        run: |
+          echo "[Sealights] Downloading Sealights Golang & CLI Agents..."
+          case $(lscpu | awk '/Architecture:/{print $2}') in
+            x86_64) SL_ARCH="linux-amd64";;
+            arm) SL_ARCH="linux-arm64";;
+          esac
+          wget -nv -O sealights-go-agent.tar.gz https://agents.sealights.co/slgoagent/latest/slgoagent-$SL_ARCH.tar.gz
+          wget -nv -O sealights-slcli.tar.gz https://agents.sealights.co/slcli/latest/slcli-$SL_ARCH.tar.gz
+          tar -xzf ./sealights-go-agent.tar.gz && tar -xzf ./sealights-slcli.tar.gz
+          rm -f ./sealights-go-agent.tar.gz ./sealights-slcli.tar.gz
+          ./slgoagent -v 2> /dev/null | grep version && ./slcli -v 2> /dev/null | grep version
+      
+      - name: Write SeaLights token into file
+        run: echo "${SEALIGHTS_AGENT_TOKEN}" > sltoken.txt
+        env:
+          SEALIGHTS_AGENT_TOKEN: '${{secrets.SEALIGHTS_AGENT_TOKEN}}' ## Make sure to add token to repo secrets
```

Comment:
I think secrets will not be accessible to PRs coming from forks. This is a security measure to prevent those PRs from potentially exfiltrating the repo secrets.

Reply:
in order to run workflows with access to repository secrets you'd need to use `pull_request_target` type of github workflow, like we do it e.g. in [konflux-ui repo](https://github.com/konflux-ci/konflux-ui/blob/62bab193d2021facbe676b687ccb35a35b1041b9/.github/workflows/pr-check.yaml#L4)

Reply:
This Workflow is still checking out potentially unsafe code and will have access to the repo secrets. We need some safeguards here. See what we are doing with an `authorize` job that requires manual approval for PRs coming from forks: https://github.com/redhat-developer/rhdh-operator/blob/main/.github/workflows/pr-container-build.yaml#L3-L41